internal/settings/settings.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package settings
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/spf13/viper"
+)
+
+const settingsFile = "flow-cli.settings"
+const settingsDir = "flow-cli"
+const settingsType = "yaml"
+
+// viperLoaded only load settings file once
+var viperLoaded = false
+
+func init() {
```

Comment:
keep in mind this is probably one of the rarest cases of init being used, generally init should be avoided in Go.